smartcosmos-extension-api/src/main/java/net/smartcosmos/platform/api/dao/IAbstractNamedObjectDAO.java

```diff
@@ -0,0 +1,39 @@
+package net.smartcosmos.platform.api.dao;
+
+/*
+ * *#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*
+ * SMART COSMOS Extension API
+ * ===============================================================================
+ * Copyright (C) 2013 - 2015 Smartrac Technology Fletcher, Inc.
+ * ===============================================================================
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+ */
+
+import net.smartcosmos.model.base.INamedObject;
+import net.smartcosmos.model.context.IAccount;
+
+import java.util.Collection;
+
+/**
+ * Initially created by tcross on November 16, 2015.
+ */
+public interface IAbstractNamedObjectDAO <U extends INamedObject<U>, V extends U> extends IAbstractDAO<U, V>
+
+{
+
+    Collection<U> findByNameLike(Class<?> clazz, String nameLike, IAccount account);
```

Comment:
They won't have access to the Class, so this needs to not be in the interface, and then appropriately utilize getEntityClass() in the implementation.  That needs to be changed everywhere else it's shown.